src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStackAllocInitializerTests.cs

```diff
@@ -2,13 +2,91 @@
 
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     [CompilerTrait(CompilerFeature.StackAllocInitializer)]
     public class CodeGenStackAllocInitializerTests : CompilingTestBase
     {
+        [Fact]
+        [WorkItem(29092, "https://github.com/dotnet/roslyn/issues/29092")]
+        public void TestMixedWithInitBlock()
+        {
+
+            var text = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        MakeBlock(1, 2, 3);
+    }
+
+    static unsafe void MakeBlock(int a, int b, int c)
+    {
+        int* ptr = stackalloc int[]
```

Comment:
Perhaps it is worth adding a test where the initializer is executed more than once due to backward branches. - Just to catch regressions in case zeroing out is accidentally "optimized".

Reply:
you mean two stackalloc init in a block? each init is executed in isolation, how could it affect zeroing out?

Reply:
Oh, you are right. Every initialization is a new allocation. Therefore it is irrelevant if control flow goes through initialization more than once.